theano/tensor/extra_ops.py

```diff
@@ -647,9 +647,36 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    if isinstance(repeats, (int, long, numpy.integer, list)):
+        repeats = numpy.asarray(repeats)
 
+    if repeats.ndim > 1:
+        raise ValueError('The dimension of repeats should not exceed 1.')
 
+    if repeats.ndim == 1:
+        return RepeatOp(axis=axis)(x, repeats)
+    else:
+        if axis == None:
+           axis = 0 
+           x = x.flatten()
+        else:
+            if axis >= x.ndim:
+                raise ValueError('Axis should not exceed x.ndim-1.')
+            if axis < 0:
+                axis = x.ndim+axis
+
+        shape = [x.shape[i] for i in xrange(x.ndim)]
+        shape_ = shape[:]
```

Comment:
Can you add comments to explain what you are doing? For instance, that you are going to use alloc to replicate `x` along an additional dimension, that `shape_` is the shape of that intermediate tensor, and that you modify `shape` to be the output shape? Or maybe name them `alloc_shape` and `output_shape`, or something.
I can understand what is happening here because I already know the approach, but comments would be necessary for the code to be maintainable.